Calibration/LumiAlCaRecoProducers/plugins/AlcaPCCIntegrator.cc

```diff
@@ -18,61 +18,63 @@ ________________________________________________________________**/
 #include "FWCore/Framework/interface/Frameworkfwd.h"
 #include "FWCore/Framework/interface/LuminosityBlock.h"
 #include "FWCore/Framework/interface/MakerMacros.h"
-#include "FWCore/Framework/interface/one/EDProducer.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 #include "FWCore/Utilities/interface/EDGetToken.h"
 //The class
-class AlcaPCCIntegrator
-    : public edm::one::EDProducer<edm::EndLuminosityBlockProducer, edm::one::WatchLuminosityBlocks> {
+class AlcaPCCIntegrator : public edm::global::EDProducer<edm::LuminosityBlockSummaryCache<reco::PixelClusterCounts>,
+                                                         edm::StreamCache<reco::PixelClusterCounts>,
+                                                         edm::EndLuminosityBlockProducer,
+                                                         edm::Accumulator> {
 public:
   explicit AlcaPCCIntegrator(const edm::ParameterSet&);
   ~AlcaPCCIntegrator() override = default;
 
-private:
-  void beginLuminosityBlock(edm::LuminosityBlock const& lumiSeg, const edm::EventSetup& iSetup) override;
-  void endLuminosityBlock(edm::LuminosityBlock const& lumiSeg, const edm::EventSetup& iSetup) override;
-  void endLuminosityBlockProduce(edm::LuminosityBlock& lumiSeg, const edm::EventSetup& iSetup) override;
-  void produce(edm::Event& iEvent, const edm::EventSetup& iSetup) override;
+  std::unique_ptr<reco::PixelClusterCounts> beginStream(edm::StreamID) const override;
+  std::shared_ptr<reco::PixelClusterCounts> globalBeginLuminosityBlockSummary(edm::LuminosityBlock const&,
+                                                                              edm::EventSetup const&) const override;
+  void accumulate(edm::StreamID iID, const edm::Event& iEvent, const edm::EventSetup&) const override;
+  void streamEndLuminosityBlockSummary(edm::StreamID,
+                                       edm::LuminosityBlock const&,
+                                       edm::EventSetup const&,
+                                       reco::PixelClusterCounts*) const override;
+  void globalEndLuminosityBlockSummary(edm::LuminosityBlock const&,
+                                       edm::EventSetup const&,
+                                       reco::PixelClusterCounts* iCounts) const override;
+  void globalEndLuminosityBlockProduce(edm::LuminosityBlock& iLumi,
+                                       edm::EventSetup const&,
+                                       reco::PixelClusterCounts const* iCounts) const override;
 
+private:
+  edm::EDPutTokenT<reco::PixelClusterCounts> lumiPutToken_;
```

Comment:
In the forthcoming PR in which you fix the trackedness of the parameters, please also make these class members `const`